src/components/site/tableOfContents/index.jsx

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React from 'react';
+import reactn from 'reactn';
+import { Link, graphql, StaticQuery } from 'gatsby';
+import PropTypes from 'prop-types';
+import { Drawer } from 'react-md';
+import styles from './table-of-contents.module.scss';
+
+// import Footer from '../components/site/footer';
+const TO_PREFIX = '/';
+const routes = [
+  {
+    label: 'home',
+    to: `${TO_PREFIX}`,
+    icon: 'home',
+    exact: 'true',
+    primaryText: 'Home',
+    active: false,
+  },
+  {
+    label: 'style-guide',
+    to: `${TO_PREFIX}StyleGuide`,
+    icon: 'style',
+    primaryText: 'Style Guide',
+    active: false,
+  },
+  { divider: true },
+  {
+    primaryText: 'Table of Contents',
+    subheader: true,
+  },
+];
+
+@reactn
+class TableOfContents extends React.PureComponent {
+  // constructor(props) {
+  //   super(props);
+  //
+  // }
+
+  componentDidMount() {
+    const { navLinks } = this.props;
+    const links = routes.concat(navLinks);
+    this.navLinks = links.map(link => {
+      if (link.divider || link.subheader) return link;
+
+      if (link.to) {
+        link.component = Link;
+      }
+
+      if (link.slug) {
+        link.component = Link;
+        link.label = link.title;
+        link.to = `${TO_PREFIX}${link.slug}`;
+        link.primaryText = link.title;
+        link.active = false;
+      }
+      return link;
+    });
+  }
+
+  // componentDidUpdate() {}
+
+  // setActivePage() {}
+
+  handleVisibility = visible => {
+    const { toggleSidebar } = this.props;
+    toggleSidebar(visible);
+  };
+
+  render() {
+    const { TEMPORARY } = Drawer.DrawerTypes;
+    const { visible } = this.props;
+
+    return (
+      <Drawer
+        type={TEMPORARY}
+        className={styles.tableOfContents}
+        visible={visible}
+        onVisibilityChange={this.handleVisibility}
+        navItems={this.navLinks}
```

Comment:
per the comment above, this would now look like `navItems={this.getNavLinks(this.props.navLinks)}`.  Or something similar.  In react world, as much as possible, if I'm writing these kinds of helper methods, I try to pass in everything it needs as arguments rather than having it talk up to this.props from inside itself.  This isn't always possible/practical, but since, in this case, you'll be calling this method in `render()`, and you're already destructuring `this.props` to grab `visible`, you might as well grab `navLinks` too, and pass it into your new `getNavLinks` method.

Reply:
Final output per the previous solution:

```
  render() {
    const { TEMPORARY } = Drawer.DrawerTypes;
    const { visible } = this.props;

    return (
      <Drawer
        type={TEMPORARY}
        className={styles.tableOfContents}
        visible={visible}
        onVisibilityChange={this.handleVisibility}
        navItems={this.navLinks}
        overlay
      />
    );
  }
```